### PR TITLE
Add Script for arbitrary asm generation

### PR DESCRIPTION
**Pull request by gianbelinche**
Closes https://github.com/lambdaclass/era_vm/issues/27
This PR adds a python script for generating arbitrary asm files.
It should be invoked like this:

`python3 generate_asm.py <path_to_your_asm_file> <output_dir>`

Or from the root directory of this project

`./generate_asm.sh <path_to_your_asm_file> <output_dir>`

The asm given should have the following format:

```nasm
	.text
	.file	"test.zasm"
	.globl	__entry

__entry:
.func_begin0:
	add r0,r0,stack+=[5]
	; %2=r9,stack[1]
	add 5, r0, %2
	; %2=r9,stack[1] ; %3=r11,stack[3]
	add %2, r0, %3
	; %3=r11,stack[3]
	sstore	r0, %3
	ret
.func_end0:

	.note.GNU-stack
	.rodata
```
Where the `%x` , x being any integer, represents the position in the file that you want to change
Let's take as an example this two lines
```nasm
        ; %2=r9,stack[1]
	add 5, r0, %2
```
Here `%2` in the second line is whats going to be replaced, and the comment (marked with ; ) tells you what is going to be replaced for.
In this case, there are 2 values `r9` and `stack[1]` separated by a comma, this means two different files are going to be generated, one with `r9` and another with `stack[1]`.
The same happens with any other `%x`, and if there are several instances of the same `%x` they don't generate extra files, this means, in the following example:
```nasm
        ; %2=r9,stack[1]
	add 5, r0, %2
        ; %2=r9,stack[1]
	add %2, r0, r11
```
Only 2 files will be generated:
```nasm
        ; %2=r9,stack[1]
	add 5, r0, r9
        ; %2=r9,stack[1]
	add r9, r0, r11
```
And
```nasm
        ; %2=r9,stack[1]
	add 5, r0, stack[1]
        ; %2=r9,stack[1]
	add stack[1], r0, r11
```

### CODE REVIEW

File: scripts/generate_asm.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+This is a python script for generating arbitrary asm files.
```
Comment (fkrause98) [inline scripts/generate_asm.md:1]: Let's rename this file to README.md for now, so GitHub picks it up automatically.

Comment (fkrause98) [inline scripts/generate_asm.md:1]: Let's rename this file to README.md for now, so GitHub picks it up automatically.

File: scripts/generate_asm.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    for str_to_replace in replacements:
+        new_files_to_replace = []
+        for _file in files_to_replace:
+            new_files = replace_once(_file,str_to_replace)
+            new_files_to_replace.extend(new_files)
+        if len(new_files_to_replace) != 0:
+            files_to_replace = new_files_to_replace
+
+    for j,_file in enumerate(files_to_replace):
+        with open(file[:-5] + "_replaced_" + str(j) + ".zasm", 'w') as f:
```
Comment (fkrause98) [inline scripts/generate_asm.py:22]: Check this slice, it's generating these files for me:
```
  exampl_replaced_1.zasm
  exampl_replaced_2.zasm
  exampl_replaced_3.zasm
```

Mind the missing 'e', I was using the example.asm file from the readme.

File: scripts/generate_asm.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import sys
```
Comment (fkrause98) [inline scripts/generate_asm.py:1]: Can we add a make target or bash script that receives a file name and uses this script to generate a test file? 

Some ideas here: https://stackoverflow.com/questions/2214575/passing-arguments-to-make-run

File: scripts/generate_asm.py (hunk 2 above)
Comment (fkrause98) [inline scripts/generate_asm.py:22]: Let's make it so the script generates the files into the tests programs folder, or accept an argument with the output folder, whichever you prefer.


Ideally, we should be able to just do make 'test_template.asm' and generate the tests files.

**Review by fkrause98 — CHANGES_REQUESTED** [review]
Looks 95% there, just some things to fix and some improvements.

Great job overall, ping me when you're done and I'll review it again.

**Review by juan518munoz — APPROVED** [review]
LGTM

**Review by fkrause98 — CHANGES_REQUESTED** [review]
The script is failing for me, please double check, here's how to reproduce it:
1. `mkdir test`
2. Copy the provided example from the PR description to 'example.zasm'.
3. Run the script from the project root: ./generate_asm.sh  example.asm test.
It should fail with an output like this:
```
fran@Franciscos-MBP ~/P/era_vm (arbitrary-tests)> ./generate_asm.sh  example.asm test
Traceback (most recent call last):
  File "/Users/fran/Programming/era_vm/scripts/generate_asm.py", line 74, in <module>
    main()
  File "/Users/fran/Programming/era_vm/scripts/generate_asm.py", line 71, in main
    replace_asm(sys.argv[1],sys.argv[2])
  File "/Users/fran/Programming/era_vm/scripts/generate_asm.py", line 21, in replace_asm
    file_name = file.rsplit(".",1)[0].rsplit("/",1)[1]
IndexError: list index out of range
```

Ping me back if you can't reproduce this error.

**Review by fkrause98 — APPROVED** [review]
LGTM!

**Pull request merged**